### ISSUE TITLE
A/F: `NumElts > 1 && "Expected at least 2-element fixed length vector(s)."' after 59a67ea

### ISSUE THREAD

**Issue by dyung** [llvm/llvm-project#67774]
Several of our internal tests started hitting an assertion failure after commit 59a67ea.

I was able to reduce one of the failures to the following code:
```c++
struct {
  long alpha : 38;
} bravo;
__attribute__((__vector_size__(sizeof(long)))) long volatile charlie;
void delta() { bravo.alpha = long(charlie); }
```
When compiled with optimizations enabled by a compiler that contains 59a67ea, you hit the assertion failure:
```
$ ~/src/upstream/59a67ea35d60-linux/bin/clang -c -O2 repro.cpp
clang: /home/dyung/src/upstream/llvm_clean_git/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:7095: llvm::InstructionCost llvm::slpvectorizer::BoUpSLP::ShuffleCostEstimator::computeExtractCost(llvm::ArrayRef<llvm::Value*>, llvm::ArrayRef<int>, llvm::
TargetTransformInfo::ShuffleKind): Assertion `NumElts > 1 && "Expected at least 2-element fixed length vector(s)."' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:        
0.      Program arguments: /home/dyung/src/upstream/59a67ea35d60-linux/bin/clang -c -O2 repro.cpp
1.      <eof> parser at end of file
2.      Optimizer   
 #0 0x000055e108d3860f llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x3cc960f)
 #1 0x000055e108d3618c llvm::sys::CleanupOnSignal(unsigned long) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x3cc718c)
 #2 0x000055e108c79e98 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #3 0x00007f3f83f46420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #4 0x00007f3f83a1300b raise /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:51:1
 #5 0x00007f3f839f2859 abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:81:7
 #6 0x00007f3f839f2729 get_sysdep_segment_value /build/glibc-SzIz7B/glibc-2.31/intl/loadmsgcat.c:509:8
 #7 0x00007f3f839f2729 _nl_load_domain /build/glibc-SzIz7B/glibc-2.31/intl/loadmsgcat.c:970:34
 #8 0x00007f3f83a03fd6 (/lib/x86_64-linux-gnu/libc.so.6+0x33fd6)
 #9 0x000055e10a49040f llvm::slpvectorizer::BoUpSLP::ShuffleCostEstimator::computeExtractCost(llvm::ArrayRef<llvm::Value*>, llvm::ArrayRef<int>, llvm::TargetTransformInfo::ShuffleKind) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x542140f)
#10 0x000055e10a491841 llvm::slpvectorizer::BoUpSLP::ShuffleCostEstimator::adjustExtracts(llvm::slpvectorizer::BoUpSLP::TreeEntry const*, llvm::ArrayRef<int>, llvm::TargetTransformInfo::ShuffleKind) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/cla
ng+0x5422841)
#11 0x000055e10a4cd05e llvm::slpvectorizer::BoUpSLP::getEntryCost(llvm::slpvectorizer::BoUpSLP::TreeEntry const*, llvm::ArrayRef<llvm::Value*>, llvm::SmallPtrSetImpl<llvm::Value*>&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x545e05e)
#12 0x000055e10a4e82b3 llvm::slpvectorizer::BoUpSLP::getTreeCost(llvm::ArrayRef<llvm::Value*>) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x54792b3)
#13 0x000055e10a4efe8f llvm::SLPVectorizerPass::tryToVectorizeList(llvm::ArrayRef<llvm::Value*>, llvm::slpvectorizer::BoUpSLP&, bool) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x5480e8f)
#14 0x000055e10a4f28b8 llvm::SLPVectorizerPass::tryToVectorize(llvm::Instruction*, llvm::slpvectorizer::BoUpSLP&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x54838b8)
#15 0x000055e10a4f2a93 llvm::SLPVectorizerPass::tryToVectorize(llvm::ArrayRef<llvm::WeakTrackingVH>, llvm::slpvectorizer::BoUpSLP&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x5483a93)
#16 0x000055e10a4f332c llvm::SLPVectorizerPass::vectorizeRootInstruction(llvm::PHINode*, llvm::Instruction*, llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&, llvm::TargetTransformInfo*) (.constprop.0) SLPVectorizer.cpp:0:0
#17 0x000055e10a4f6526 llvm::SLPVectorizerPass::vectorizeChainsInBlock(llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x5487526)
#18 0x000055e10a4f9c16 llvm::SLPVectorizerPass::runImpl(llvm::Function&, llvm::ScalarEvolution*, llvm::TargetTransformInfo*, llvm::TargetLibraryInfo*, llvm::AAResults*, llvm::LoopInfo*, llvm::DominatorTree*, llvm::AssumptionCache*, llvm::DemandedBits*
, llvm::OptimizationRemarkEmitter*) (.part.0) SLPVectorizer.cpp:0:0
#19 0x000055e10a4fa5aa llvm::SLPVectorizerPass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x548b5aa)
#20 0x000055e10a0627f6 llvm::detail::PassModel<llvm::Function, llvm::SLPVectorizerPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/home/dyung/src/upstream/59a67ea35d6
0-linux/bin/clang+0x4ff37f6)
#21 0x000055e1065283bd llvm::detail::PassModel<llvm::Function, llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::
Function>&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x14b93bd)
#22 0x000055e1086decfd llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x366fcfd)
#23 0x000055e10650b446 llvm::detail::PassModel<llvm::Module, llvm::ModuleToFunctionPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/dyung/src/upstream/59a67ea35
d60-linux/bin/clang+0x149c446)
#24 0x000055e1086dcc7d llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x366dc7d)
#25 0x000055e108fb70ff (anonymous namespace)::EmitAssemblyHelper::RunOptimizationPipeline(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>&, std::unique_ptr<llvm::ToolOutputFile, std::default
_delete<llvm::ToolOutputFile>>&) BackendUtil.cpp:0:0
#26 0x000055e108fba4ef (anonymous namespace)::EmitAssemblyHelper::EmitAssembly(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>) BackendUtil.cpp:0:0
#27 0x000055e108fbab39 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, l
lvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x3f4bb39)
#28 0x000055e109ffc9c7 clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x4f8d9c7)
#29 0x000055e10b1feb89 clang::ParseAST(clang::Sema&, bool, bool) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x618fb89)
#30 0x000055e109ffb128 clang::CodeGenAction::ExecuteAction() (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x4f8c128)
#31 0x000055e109839309 clang::FrontendAction::Execute() (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x47ca309)
#32 0x000055e1097b7bde clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x4748bde)
#33 0x000055e10991cc56 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x48adc56)
#34 0x000055e106150546 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x10e1546)
#35 0x000055e106147c1a ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&, llvm::ToolContext const&) driver.cpp:0:0
#36 0x000055e10960cf8d void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const::'lam
bda'()>(long) Job.cpp:0:0
#37 0x000055e108c7a3a0 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x3c0b3a0)
#38 0x000055e10960d5af clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (.part.0) Job.cpp:0:0
#39 0x000055e1095d43bc clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x45653bc)                                             
#40 0x000055e1095d4e4d clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x4565e4d)
#41 0x000055e1095dd3e5 clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x456e3e5)
#42 0x000055e10614d92a clang_main(int, char**, llvm::ToolContext const&) (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x10de92a)                                                                                                                 
#43 0x000055e106063265 main (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0xff4265)
#44 0x00007f3f839f4083 __libc_start_main /build/glibc-SzIz7B/glibc-2.31/csu/../csu/libc-start.c:342:3                                                                                                                                                      
#45 0x000055e1061476de _start (/home/dyung/src/upstream/59a67ea35d60-linux/bin/clang+0x10d86de)                                                                                                                                                            
clang: error: clang frontend command failed with exit code 134 (use -v to see invocation)
clang version 18.0.0 (https://github.com/llvm/llvm-project.git 59a67ea35d608480257fc64ec3e5106ef50de740)                                                                                                                                                   
Target: x86_64-unknown-linux-gnu                                                                                                                                                                                                                           
Thread model: posix                                                                                                                                                                                                                                        
InstalledDir: /home/dyung/src/upstream/59a67ea35d60-linux/bin
```
@alexey-bataev this was your change, can you take a look?

**Comment by Endilll** [comment]
Seems quite a recent regression, because it doesn't reproduce with 17: https://godbolt.org/z/vPqh9oqGK
```
clang++: /root/llvm-project/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:7095:
llvm::InstructionCost llvm::slpvectorizer::BoUpSLP::ShuffleCostEstimator::computeExtractCost(llvm::ArrayRef<llvm::Value*>, llvm::ArrayRef<int>, llvm::TargetTransformInfo::ShuffleKind):
Assertion `NumElts > 1 && "Expected at least 2-element fixed length vector(s)."' failed.

PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /opt/compiler-explorer/clang-assertions-trunk/bin/clang++ -gdwarf-4 -g -o /app/output.s -mllvm --x86-asm-syntax=intel -S --gcc-toolchain=/opt/compiler-explorer/gcc-snapshot -fcolor-diagnostics -fno-crash-diagnostics -O3 <source>
1.	<eof> parser at end of file
2.	Optimizer
 #0 0x000000000370a088 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x370a088)
 #1 0x0000000003707d4c llvm::sys::CleanupOnSignal(unsigned long) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x3707d4c)
 #2 0x00000000036509a8 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #3 0x00007f03080db420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #4 0x00007f0307b9e00b raise (/lib/x86_64-linux-gnu/libc.so.6+0x4300b)
 #5 0x00007f0307b7d859 abort (/lib/x86_64-linux-gnu/libc.so.6+0x22859)
 #6 0x00007f0307b7d729 (/lib/x86_64-linux-gnu/libc.so.6+0x22729)
 #7 0x00007f0307b8efd6 (/lib/x86_64-linux-gnu/libc.so.6+0x33fd6)
 #8 0x0000000004f5493f llvm::slpvectorizer::BoUpSLP::ShuffleCostEstimator::computeExtractCost(llvm::ArrayRef<llvm::Value*>, llvm::ArrayRef<int>, llvm::TargetTransformInfo::ShuffleKind) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4f5493f)
 #9 0x0000000004f55d31 llvm::slpvectorizer::BoUpSLP::ShuffleCostEstimator::adjustExtracts(llvm::slpvectorizer::BoUpSLP::TreeEntry const*, llvm::ArrayRef<int>, llvm::TargetTransformInfo::ShuffleKind) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4f55d31)
#10 0x0000000004f904fd llvm::slpvectorizer::BoUpSLP::getEntryCost(llvm::slpvectorizer::BoUpSLP::TreeEntry const*, llvm::ArrayRef<llvm::Value*>, llvm::SmallPtrSetImpl<llvm::Value*>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4f904fd)
#11 0x0000000004faafd3 llvm::slpvectorizer::BoUpSLP::getTreeCost(llvm::ArrayRef<llvm::Value*>) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4faafd3)
#12 0x0000000004fb2b4f llvm::SLPVectorizerPass::tryToVectorizeList(llvm::ArrayRef<llvm::Value*>, llvm::slpvectorizer::BoUpSLP&, bool) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4fb2b4f)
#13 0x0000000004fb546c llvm::SLPVectorizerPass::tryToVectorize(llvm::Instruction*, llvm::slpvectorizer::BoUpSLP&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4fb546c)
#14 0x0000000004fb56d3 llvm::SLPVectorizerPass::tryToVectorize(llvm::ArrayRef<llvm::WeakTrackingVH>, llvm::slpvectorizer::BoUpSLP&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4fb56d3)
#15 0x0000000004fb5eec llvm::SLPVectorizerPass::vectorizeRootInstruction(llvm::PHINode*, llvm::Instruction*, llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&, llvm::TargetTransformInfo*) (.constprop.0) SLPVectorizer.cpp:0:0
#16 0x0000000004fb909e llvm::SLPVectorizerPass::vectorizeChainsInBlock(llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4fb909e)
#17 0x0000000004fbc6fd llvm::SLPVectorizerPass::runImpl(llvm::Function&, llvm::ScalarEvolution*, llvm::TargetTransformInfo*, llvm::TargetLibraryInfo*, llvm::AAResults*, llvm::LoopInfo*, llvm::DominatorTree*, llvm::AssumptionCache*, llvm::DemandedBits*, llvm::OptimizationRemarkEmitter*) (.part.0) SLPVectorizer.cpp:0:0
#18 0x0000000004fbd06a llvm::SLPVectorizerPass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4fbd06a)
#19 0x0000000004b4790e llvm::detail::PassModel<llvm::Function, llvm::SLPVectorizerPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4b4790e)
#20 0x0000000000fa43bd llvm::detail::PassModel<llvm::Function, llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0xfa43bd)
#21 0x00000000030d636e llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x30d636e)
#22 0x0000000000f8878e llvm::detail::PassModel<llvm::Module, llvm::ModuleToFunctionPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0xf8878e)
#23 0x00000000030d436c llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x30d436c)
#24 0x000000000397dc40 (anonymous namespace)::EmitAssemblyHelper::RunOptimizationPipeline(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>&, std::unique_ptr<llvm::ToolOutputFile, std::default_delete<llvm::ToolOutputFile>>&) BackendUtil.cpp:0:0
#25 0x0000000003980f86 (anonymous namespace)::EmitAssemblyHelper::EmitAssembly(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>) BackendUtil.cpp:0:0
#26 0x0000000003981599 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem>, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x3981599)
#27 0x00000000049762bf clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x49762bf)
#28 0x0000000005e71b09 clang::ParseAST(clang::Sema&, bool, bool) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x5e71b09)
#29 0x0000000004974a68 clang::CodeGenAction::ExecuteAction() (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x4974a68)
#30 0x00000000041d7bd9 clang::FrontendAction::Execute() (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x41d7bd9)
#31 0x000000000415898e clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x415898e)
#32 0x00000000042b6dce clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x42b6dce)
#33 0x0000000000bdcd16 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0xbdcd16)
#34 0x0000000000bd45da ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&, llvm::ToolContext const&) driver.cpp:0:0
#35 0x0000000003fb6099 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const::'lambda'()>(long) Job.cpp:0:0
#36 0x0000000003650e54 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x3650e54)
#37 0x0000000003fb668f clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (.part.0) Job.cpp:0:0
#38 0x0000000003f7e9d5 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x3f7e9d5)
#39 0x0000000003f7f43d clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x3f7f43d)
#40 0x0000000003f87365 clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0x3f87365)
#41 0x0000000000bda1bc clang_main(int, char**, llvm::ToolContext const&) (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0xbda1bc)
#42 0x0000000000ad4f31 main (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0xad4f31)
#43 0x00007f0307b7f083 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24083)
#44 0x0000000000bd40be _start (/opt/compiler-explorer/clang-assertions-trunk/bin/clang+++0xbd40be)
clang++: error: clang frontend command failed with exit code 134 (use -v to see invocation)
```

**Comment by Endilll** [comment]
CC @alexey-bataev as author of the aforementioned commit.

**Comment by alexey-bataev** [comment]
Must be fixed with 019aee832768b0704048905cd8ce20299e7e0b25
